### PR TITLE
Add create command to keepassxc-cli

### PR DESCRIPTION
**Pull request by louib**
Rebasing https://github.com/keepassxreboot/keepassxc/pull/2352
(credits to @vi-n)

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)
- ✅ Documentation (non-code change)

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by louib** [comment]
@phoerious @droidmonkey I updated this PR with the recent changes in the CLI  module (Using CLI streams like `Utils::STDOUT`, using global options like `Command::KeyFileOption`). I think all that's missing is some unit tests, and maybe the `-s` option too.

**Comment by louib** [comment]
@droidmonkey I added 2 tests, but would like to add 1 more before merging (especially testing the key file creation).

**Comment by louib** [comment]
@droidmonkey should be good now, added unit tests for `-k` option. Would you mind doing a quick test run on Windows before merging? Again, not sure why the Windows build is failing (`SSH exit-code 1`)

**Comment by droidmonkey** [comment]
Ok I'll take a look at some point

File: src/cli/Create.cpp [hunk 1]
```diff
@@ -132,7 +132,7 @@ QSharedPointer<PasswordKey> Create::getPasswordFromStdin()
     out.flush();
     QString password = Utils::getPassword();
 
-    if (password.length() > 0) {
+    if (password.isEmpty()) {
```
Comment (louib) [inline src/cli/Create.cpp]: @droidmonkey shouldn't that be `!password.isEmpty()`??

Comment (droidmonkey) [reply src/cli/Create.cpp]: yes sir! I need to fix tests too

**Pull request merged**